browser/ui/views/side_panel/brave_side_panel_view_base.h

```diff
@@ -7,10 +7,19 @@
 #define BRAVE_BROWSER_UI_VIEWS_SIDE_PANEL_BRAVE_SIDE_PANEL_VIEW_BASE_H_
 
 #include "base/scoped_observation.h"
+#include "ui/native_theme/native_theme.h"
+#include "ui/native_theme/native_theme_observer.h"
 #include "ui/views/view.h"
 #include "ui/views/view_observer.h"
 
-class BraveSidePanelViewBase : public views::View, public views::ViewObserver {
+// Observe native theme changes to align with panel web contents theme.
+// Child view's OnThemeChanged() is not called for private/tor window
+// whenever brave theme is changed because these window uses dark theme
+// provider always. To get brave theme change notification, we need to
+// observe native theme here.
```

Comment:
I'm wondering if this could happen to other UI components too. If then, could it be better to make `BrowserRootView` observe the NativeTheme, and call `RootView::ThemeChanged()` in case it's private/tor window, instead of subclassings in this PR.

Reply:
Great advice! With `ThemeChanged()`, PR is more simplified. PTAL again.